comms/dht/src/peer_validator.rs

```diff
@@ -67,55 +67,55 @@ impl<'a> PeerValidator<'a> {
-                    // added date
-                    current_peer
-                        .identity_signature
-                        .as_ref()
-                        .map(|i| i.updated_at() < new_dt && (
-                            !current_peer.is_seed() ||
-                            current_peer.added_at < new_dt.naive_utc()))
-                        // If None, update to peer with valid signature
-                        .unwrap_or(true)
-                };
-
-                if !can_update {
-                    debug!(
-                        target: LOG_TARGET,
-                        "Peer `{}` already exists or is up to date and will not be updated", new_peer.node_id
-                    );
-                    return Ok(false);
-                }
+            Some(current_peer) => {
+                // let can_update = can_update && {
+                //     // Update/insert peer if newer
+                //     // unreachable panic: can_update is true only is identity_signature is present and valid
+                //     // let new_dt = new_peer
+                //     //     .identity_signature
+                //     //     .as_ref()
+                //     //     .map(|i| i.updated_at())
+                //     //     .expect("unreachable panic");
+                //
+                //     // Update if new_peer has newer timestamp than current_peer, and if the newer timestamp is after
+                // the     // added date
+                //     current_peer
+                //         .identity_signature
+                //         .as_ref()
+                //         .map(|i| i.updated_at() < new_dt && (
+                //             !current_peer.is_seed() ||
+                //             current_peer.added_at < new_dt.naive_utc()))
+                //         // If None, update to peer with valid signature
+                //         .unwrap_or(true)
+                // };
+
+                // if !can_update {
+                //     debug!(
+                //         target: LOG_TARGET,
+                //         "Peer `{}` already exists or is up to date and will not be updated", new_peer.node_id
+                //     );
+                //     return Ok(false);
+                // }
 
                 debug!(target: LOG_TARGET, "Updating peer `{}`", new_peer.node_id);
-                current_peer
-                    .update_addresses(new_peer.addresses.into_vec())
-                    .set_features(new_peer.features)
-                    .set_offline(false);
-                if let Some(sig) = new_peer.identity_signature {
-                    current_peer.set_valid_identity_signature(sig);
-                }
+                // current_peer
+                //     .update_addresses(new_peer.addresses.into_vec())
+                //     .set_features(new_peer.features);
+                // if let Some(sig) = new_peer.identity_signature {
```

Comment:
I dont think we should remove this

Reply:
See the notes in the motivation about this